demisto_sdk/commands/common/GitContentConfig.py

```diff
@@ -12,60 +18,77 @@
 from demisto_sdk.commands.common.git_util import GitUtil
 
 
+class GitProvider(enum.Enum):
+    GitHub = 'github'
+    GitLab = 'gitlab'
+
+
 class GitCredentials:
     ENV_GITHUB_TOKEN_NAME = 'DEMISTO_SDK_GITHUB_TOKEN'
     ENV_GITLAB_TOKEN_NAME = 'DEMISTO_SDK_GITLAB_TOKEN'
-    GITHUB_TOKEN: Optional[str]
-    GITLAB_TOKEN: Optional[str]
 
     def __init__(self):
-        self.GITHUB_TOKEN = os.getenv(self.ENV_GITHUB_TOKEN_NAME)
-        self.GITLAB_TOKEN = os.getenv(self.ENV_GITLAB_TOKEN_NAME)
+        self.github_token = os.getenv(self.ENV_GITHUB_TOKEN_NAME)
+        self.gitlab_token = os.getenv(self.ENV_GITLAB_TOKEN_NAME)
 
 
 class GitContentConfig:
     """Holds links, credentials and other content related github configuration
 
     Attributes:
-        CURRENT_REPOSITORY: The current repository in the cwd
-        CONTENT_GITHUB_LINK: Link to the raw content git repository
-        CONTENT_GITHUB_MASTER_LINK: Link to the content git repository's master branch
-        Credentials: Credentials to the git.
+        credentials: Credentials to the git.
     """
-    BASE_RAW_GITHUB_LINK = r'https://raw.githubusercontent.com/'
+    BASE_RAW_GITHUB_LINK = r'https://raw.{GITHUB_HOST}/'
     SDK_API_GITHUB_RELEASES = r'https://api.github.com/repos/demisto/demisto-sdk/releases'
     OFFICIAL_CONTENT_REPO_NAME = 'demisto/content'
     CONTENT_GITHUB_UPSTREAM = r'upstream.*demisto/content'
     CONTENT_GITHUB_ORIGIN = r'origin.*demisto/content'
+    GITHUB_USER_CONTENT = 'githubusercontent.com'
 
-    CURRENT_REPOSITORY: str
-    CONTENT_GITHUB_LINK: str
-    CONTENT_GITHUB_MASTER_LINK: str
-
-    GITLAB_HOST: Optional[str] = None
-    GITLAB_ID: Optional[int] = None
     BASE_RAW_GITLAB_LINK = "https://{GITLAB_HOST}/api/v4/projects/{GITLAB_ID}/repository"
 
-    def __init__(self, repo_name: Optional[str] = None):
-        self.Credentials = GitCredentials()
-        if not repo_name:
+    ENV_REPO_HOSTNAME_NAME = 'DEMISTO_SDK_REPO_HOSTNAME'
+
+    def __init__(self, repo_name: Optional[str] = None, git_provider: Optional[GitProvider] = None, repo_hostname: Optional[str] = None):
```

Comment:
```suggestion
    def __init__(self, repo_name: Optional[str] = None, git_provider: GitProvider = GitProvider.github, repo_hostname: Optional[str] = None):
```
Why not just give it as a default?
Then you can delete the fallback to github